youtube_dl/utils.py

```diff
@@ -881,7 +881,42 @@ def _create_http_connection(ydl_handler, http_class, is_https, *args, **kwargs):
         kwargs['strict'] = True
     hc = http_class(*args, **compat_kwargs(kwargs))
     source_address = ydl_handler._params.get('source_address')
+
     if source_address is not None:
+        filter_for = socket.AF_INET
+        if '.' not in source_address:
+            filter_for = socket.AF_INET6
```

Comment:
`filter_for = socket.AF_INET if '.' in source_address else socket.AF_INET6`.

Reply:
I made the fixes in [this pull request #17362](https://github.com/rg3/youtube-dl/pull/17362) because I deleted the repository and can't connect these.